src/components/modules/banners/contributions/ContributionsBannerWrapper.tsx

```diff
@@ -99,28 +79,60 @@ const withBannerData = (
             numArticles,
             'banner',
         );
-        const mobileMessageTextWithArticleCount = !!cleanMobileMessageText
-            ? replaceArticleCount(cleanMobileMessageText, numArticles, 'banner')
-            : null;
         const highlightedTextWithArticleCount = !!cleanHighlightedText
             ? replaceArticleCount(cleanHighlightedText, numArticles, 'banner')
             : null;
 
-        if (!copyHasPlaceholder) {
+        if (copyHasPlaceholder) {
+            throw Error('Banner copy contains placeholders, abandoning.');
+        }
+
+        return {
+            cleanHighlightedText: highlightedTextWithArticleCount,
+            cleanMessageText: messageTextWithArticleCount,
+            cleanHeading: headingWithArticleCount,
+            ctaUrl,
+            ctaText: cta.text,
+        };
+    };
+
+    const onContributeClick = (): void => {
+        const componentClickEvent = createClickEventFromTracking(
+            bannerProps.tracking,
+            ctaComponentId,
+        );
+        if (submitComponentEvent) {
+            submitComponentEvent(componentClickEvent);
+        }
+    };
+
+    const onCloseClick = (): void => {
+        const componentClickEvent = createClickEventFromTracking(tracking, closeComponentId);
+        if (submitComponentEvent) {
+            submitComponentEvent(componentClickEvent);
+        }
+        onClose();
+    };
+
+    try {
+        const renderedContent =
+            content && content.cta && buildRenderedContent(content, content.cta);
+        const renderedMobileContent =
+            mobileContent &&
+            mobileContent.cta &&
+            buildRenderedContent(mobileContent, mobileContent.cta);
```

Comment:
could this function just take one argument? (also we could potentially use optional chaining?)

Reply:
Annoyingly `cta` is optional on the `BannerContent` type so it needs to be passed separately here.
I can use optional chaining though!

Reply:
ahh gotcha 👍